autoyast/provision.erb

```diff
@@ -8,7 +8,7 @@ name: AutoYaST default
   os_minor = @host.operatingsystem.minor.to_i
   # safemode renderer does not support unary negation
   pm_set = @host.puppetmaster.empty? ? false : true
-  puppet_enabled = pm_set || @host.params['force-puppet'] && @host.params['force-puppet'] == 'true'
+  puppet_enabled = pm_set || (param_true? @host.params['force-puppet'])
```

Comment:
2) you need to pass just the name of the parameter in, not the value, so `@host.param_true? 'force-puppet'`
3) style-wise I'd suggest using parentheses for the method call instead of wrapping it all, so `@host.param_true?('force-puppet')`